examples/Mixtures/slot_based.py

```diff
@@ -0,0 +1,268 @@
+## Example for Modelling a Slot-Based Mixture
+
+### Terminology
```

Comment:
After having read the full example I am wondering if this file should rather be a user guide instead of an example. I get that it would be very hard to separate the description that we have here in a user guide while still having an intact and understandable example, but I nevertheless have the feeling that most of the description is not suitable for an example. What is @AdrianSosic opinion here?

Reply:
@AVHopp please take a look at the constraints user guide, much of this content here is already there (just not used in a single consistent use case but rather a small example per constraint), so imo its kind of there already

what exactly do you want to shift to the user guide?

Reply:
I also felt that the general "slot-based vs. traditional" debate should be move to a more central place, not the examples. But I don't necessarily see this as part of the current PR, because I can't imagine a good spot for it at the moment. We can perhaps consider creating some form of "Recipies" section in the future, i.e. stuff that contains useful practices but that is not necessarily userguide content 🤔

Reply:
ngl, to me `examples` is `recipes`, especially when the recipe requires a larger context -> This is the case here, a mixture modelling requires the combination of several baybe components

One thing is for sure: each mixture modelling method needs its separate example. You guys seem to want to pull some/much of the text into a userguide? I cant really imagine that, some of these explanations dont really make sense based on a single constraint type or so (like already done in the userguide) but require the context of the larger thing -> that userguide would eventually be very similar to this example

Reply:
We could move the general discussion on top of the header file, i.e., in `examples/Mixtures/Mixtures_Header.md`: Then, the description of the general "slot-based vs. traditional" theme is part of a more central place but still very close to the actual examples. Would this (for now) be a good compromise? (These pages could be used for such things in general, currently these are all just dummy files!)
If not, than I would say that we keep the current structure since there will be a round of restructuring examples/userguides anyway in the future, so I'd be fine with deferring this discussion to that point in time.

Reply:
I've spent way too much time on the existing examples and are no longer in the mood to make further adjustments, tbh 😄 But not blocking anyone of you to do so.

Reply:
Well, this is @Scienfitz PR, so in the end it is his decision for this PR. And you yourself mentioned several times that you would like a refactoring of examples @AdrianSosic 😉

Reply:
I think that suggestion is wortha try, although I think there is a danger hta this then will get overlooked

personally I think the explanation of the differences betweent he mixture representations is much more relevant than the lengthy excursion on combinatorics that has been added :D

Reply:
I also see the danger that it gets overlooked since people probably don't go to these pages at all ...

Reply:
Ok, then let's not move it to the other page now (although this might be something interesting for the future), since I also agree with the danger of this not being seen there. Then I'm voting for "Let's keep everything as-is to get this finally done"